velox/functions/prestosql/BinaryFunctions.h

```diff
@@ -436,4 +437,31 @@ struct FromIEEE754Bits32 {
   }
 };
 
+/// lpad(binary, padbinary, size) -> varbinary
+///     Left pads input to size characters with padding.  If size is
+///     less than the length of input, the result is truncated to size
+///     characters.  size must not be negative and padding must be non-empty.
+/// rpad(binary, padbinary, size) -> varbinary
+///     Right pads input to size characters with padding.  If size is
+///     less than the length of input, the result is truncated to size
+///     characters.  size must not be negative and padding must be non-empty.
+template <typename T, bool lpad>
+struct PadFunctionVarbinaryBase {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Varbinary>& result,
+      const arg_type<Varbinary>& binary,
+      const arg_type<Varbinary>& padbinary,
+      const arg_type<int64_t>& size) {
+    stringImpl::pad<lpad, false /*isAscii*/>(result, binary, size, padbinary);
```

Comment:
Curious, why not implement an ascii version with call_ascii , as stringImpl::pad obviously supports it ?

Reply:
@kgpai : Do we prefer to write an ascii version of the function for binary types ? I presumed those are applicable for VARCHAR only.

Reply:
@aditi-pandit Good point. I wonder if there are enough use cases where we might want to make that check on varbinaries if the function has a callascii method. I think looking at the code if there are no varchar arguments, it will still say all are ascii ? https://github.com/facebookincubator/velox/blob/6b0cb0f998595f9b7c5796cc0e33a71c1bace4c1/velox/expression/SimpleFunctionAdapter.h#L459 

I need to double check if thats a potential bug.